actor/cfnetworkingaction/policy.go

```diff
@@ -0,0 +1,184 @@
+package cfnetworkingaction
+
+import (
+	"code.cloudfoundry.org/cli/actor/v3action"
+	"code.cloudfoundry.org/cli/api/cfnetworking/cfnetv1"
+)
+
+type PolicyDoesNotExistError struct{}
+
+func (e PolicyDoesNotExistError) Error() string {
+	return "Policy does not exist."
+}
+
+type Policy struct {
+	SourceName      string
+	DestinationName string
+	Protocol        string
+	StartPort       int
+	EndPort         int
+}
+
+func (actor Actor) AddNetworkPolicy(spaceGUID, srcAppName, destAppName, protocol string, startPort, endPort int) (Warnings, error) {
+	var allWarnings Warnings
+
+	srcApp, warnings, err := actor.V3Actor.GetApplicationByNameAndSpace(srcAppName, spaceGUID)
+	allWarnings = append(allWarnings, Warnings(warnings)...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	destApp, warnings, err := actor.V3Actor.GetApplicationByNameAndSpace(destAppName, spaceGUID)
+	allWarnings = append(allWarnings, Warnings(warnings)...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	err = actor.NetworkingClient.CreatePolicies([]cfnetv1.Policy{
+		{
+			Source: cfnetv1.PolicySource{
+				ID: srcApp.GUID,
+			},
+			Destination: cfnetv1.PolicyDestination{
+				ID:       destApp.GUID,
+				Protocol: cfnetv1.PolicyProtocol(protocol),
+				Ports: cfnetv1.Ports{
+					Start: startPort,
+					End:   endPort,
+				},
+			},
+		},
+	})
+	return allWarnings, err
+}
+
+func (actor Actor) NetworkPoliciesBySpace(spaceGUID string) ([]Policy, Warnings, error) {
+	var allWarnings Warnings
+
+	applications, warnings, err := actor.V3Actor.GetApplicationsBySpace(spaceGUID)
+	allWarnings = append(allWarnings, Warnings(warnings)...)
+	if err != nil {
+		return []Policy{}, allWarnings, err
+	}
+
+	var v1Policies []cfnetv1.Policy
+	v1Policies, err = actor.NetworkingClient.ListPolicies()
+	if err != nil {
+		return []Policy{}, allWarnings, err
+	}
+
+	var policies []Policy
+	emptyPolicy := Policy{}
+	for _, v1Policy := range v1Policies {
+		policy := transformPolicy(applications, v1Policy)
```

Comment:
instead of calling `transformPolicy` for each policy, which loops over every app, it would be more efficient to create first a map of app GUID to app name and in current loop just get the app name by its GUID.